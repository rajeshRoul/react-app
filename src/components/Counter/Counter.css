#Counter{
    display: flex;
    align-items: center;
}

#Counter button{
    width: 40px;
    height: 40px;
    font-size: 1.5rem;
    margin: 0 10px;
}