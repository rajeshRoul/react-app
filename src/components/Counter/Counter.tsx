import React, { ReactElement, useState } from 'react'
import { createStore } from 'redux'
import './Counter.css'

function counterReducer(
    state = { value: 0 },
    action: { type: string }
): { value: number } {
    switch (action.type) {
        case 'counter/incremented':
            return { value: state.value + 1 }
        case 'counter/decremented':
            return { value: state.value - 1 }
        default:
            return state
    }
}

const store = createStore(counterReducer)

function Counter(): ReactElement {
    const [counter, setCounter] = useState(0)
    store.subscribe(() => setCounter(store.getState().value))

    const decrementCounter = (): void => {
        store.dispatch({ type: 'counter/decremented' })
    }

    const incrementCounter = (): void => {
        store.dispatch({ type: 'counter/incremented' })
    }

    return (
        <div data-testid="Counter" id="Counter">
            <button
                data-testid="decrementCounter"
                type="button"
                onClick={decrementCounter}
            >
                -
            </button>
            <h1 data-testid="counterValue">{counter}</h1>
            <button
                data-testid="incrementCounter"
                type="button"
                onClick={incrementCounter}
            >
                +
            </button>
        </div>
    )
}

export default Counter
