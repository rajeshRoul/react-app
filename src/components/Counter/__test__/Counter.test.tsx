/* eslint-disable no-undef */
import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import Counter from '../Counter'
import '@testing-library/jest-dom/extend-expect'

afterEach(cleanup)

test('increment/decrement on button clicks', () => {
    const { getByTestId } = render(<Counter />)
    expect(getByTestId('counterValue')).toHaveTextContent('0')
    fireEvent.click(getByTestId('decrementCounter'))
    expect(getByTestId('counterValue')).toHaveTextContent('-1')
    fireEvent.click(getByTestId('incrementCounter'))
    expect(getByTestId('counterValue')).toHaveTextContent('0')
    fireEvent.click(getByTestId('incrementCounter'))
    expect(getByTestId('counterValue')).toHaveTextContent('1')
})

it('should take a snapshot', () => {
    const { getByTestId } = render(<Counter />)
    expect(getByTestId('Counter')).toMatchSnapshot()
})
