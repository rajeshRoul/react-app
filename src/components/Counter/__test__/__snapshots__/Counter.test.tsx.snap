// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div
  data-testid="Counter"
  id="Counter"
>
  <button
    data-testid="decrementCounter"
    type="button"
  >
    -
  </button>
  <h1
    data-testid="counterValue"
  >
    0
  </h1>
  <button
    data-testid="incrementCounter"
    type="button"
  >
    +
  </button>
</div>
`;
